test/libsolidity/util/SoltestTypes.h

```diff
@@ -302,8 +303,35 @@ struct FunctionCall
 	/// Marks this function call as "short-handed", meaning
 	/// no `->` declared.
 	bool omitsArrow = true;
+	/// A textual representation of the expected side-effect of the function call.
+	std::vector<std::string> expectedSideEffects{};
+	/// A textual representation of the actual side-effect of the function call.
+	std::vector<std::string> actualSideEffects{};
 };
 
 using Builtin = std::function<std::optional<bytes>(FunctionCall const&)>;
+using SideEffectHook = std::function<std::vector<std::string>(FunctionCall const&)>;
+
+/// LogRecord
+struct LogRecord
+{
+	size_t index;
```

Comment:
Is `index` actually needed? You're not using it as far as I can tell and to preserve order you just store logs in a vector instead.

Reply:
That's true, `index` is not needed anymore. Initially I did the event matching via the contract abi - but I changed that part recently. However, I'm not sure how I used it exactly, but it can be removed now.